Puc/v4p2/Vcs/GitLabApi.php

```diff
@@ -0,0 +1,285 @@
+			if ( preg_match( '@^/?(?P<username>[^/]+?)/(?P<repository>[^/#?&]+?)/?$@', $path, $matches ) ) {
+				$this->userName = $matches['username'];
+				$this->repositoryName = $matches['repository'];
+			}
+			// this is not a traditional url, it could be gitlab is in a deeper subdirectory
+			else {
+				// get the path segments
+				$segments = explode( '/', untrailingslashit( ltrim( $path, '/' ) ) );
+
+				// we need atleast /user-name/repository-name/
+				if ( sizeof ( $segments ) < 2 ) {
+					throw new InvalidArgumentException( 'Invalid GitLab repository URL: "' . $repositoryUrl . '"' );
+				}
+
+				// get the username and repository name
+				$usernameRepo = array_splice( $segments, -2, 2 );
+				$this->userName = $usernameRepo[0];
+				$this->repositoryName = $usernameRepo[1];
+
+				// append the remaining segments to the host
+				$this->repositoryHost = trailingslashit( $this->repositoryHost ) . implode( '/', $segments );
+			}
+
+			parent::__construct( $repositoryUrl, $accessToken );
+		}
+
+		/**
+		 * Get the latest release from GitLab.
+		 *
+		 * @return Puc_v4p2_Vcs_Reference|null
+		 */
+		public function getLatestRelease() {
+			return $this->getLatestTag();
+		}
+
+		/**
+		 * Get the tag that looks like the highest version number.
+		 *
+		 * @return Puc_v4p2_Vcs_Reference|null
+		 */
+		public function getLatestTag() {
+			$tags = $this->api( '/:user/:repo/repository/tags' );
+			if ( is_wp_error( $tags ) || empty( $tags ) || ! is_array( $tags ) ) {
+				return null;
+			}
+
+			$versionTags = $this->sortTagsByVersion( $tags );
+			if ( empty( $versionTags ) ) {
+				return null;
+			}
+
+			$tag = $versionTags[0];
+			return new Puc_v4p2_Vcs_Reference( array(
+				'name' => $tag->name,
+				'version' => ltrim( $tag->name, 'v' ),
+				'downloadUrl' => $this->buildArchiveDownloadUrl( $tag->name ),
+				'apiResponse' => $tag
+			) );
+		}
+
+		/**
+		 * Get a branch by name.
+		 *
+		 * @param string $branchName
+		 * @return null|Puc_v4p2_Vcs_Reference
+		 */
+		public function getBranch( $branchName ) {
+			$branch = $this->api( '/:user/:repo/repository/branches/' . $branchName );
+			if ( is_wp_error( $branch ) || empty( $branch ) ) {
+				return null;
+			}
+
+			$reference = new Puc_v4p2_Vcs_Reference( array(
+				'name' => $branch->name,
+				'downloadUrl' => $this->buildArchiveDownloadUrl( $branch->name ),
+				'apiResponse' => $branch,
+			) );
+
+			if ( isset ( $branch->commit, $branch->commit->committed_date ) ) {
+				$reference->updated = $branch->commit->committed_date;
+			}
+
+			return $reference;
+		}
+
+		/**
+		 * Get the latest commit that changed the specified file.
+		 *
+		 * @param string $filename
+		 * @param string $ref Reference name (e.g. branch or tag).
+		 * @return StdClass|null
+		 */
+		public function getLatestCommit( $filename, $ref = 'master' ) {
+			throw new LogicException( 'The ' . __METHOD__ . ' method is not implemented and should not be used.' );
+		}
```

Comment:
I think you can safely remove the getLatestCommit method. It's not actually used by any PUC code since version 4.0.